engine/lib/master.go

```diff
@@ -625,3 +633,38 @@ func (m *DefaultBaseMaster) CreateWorker(
 func (m *DefaultBaseMaster) IsMasterReady() bool {
 	return m.workerManager.IsInitialized()
 }
+
+// SetProjectInfo set the project info of specific worker
+// [NOTICE]: Only used by JobManager to set project for different job(worker for jobmanager)
+func (m *DefaultBaseMaster) SetProjectInfo(workerID libModel.WorkerID, projectInfo tenant.ProjectInfo) {
+	m.workerProjectMap.Store(workerID, projectInfo)
+}
+
+// DeleteProjectInfo delete the project info of specific worker
+// NOTICEL Only used by JobMananger when stop job
+func (m *DefaultBaseMaster) DeleteProjectInfo(workerID libModel.WorkerID) {
```

Comment:
`DeleteProjectInfo` is not used now, should we remove it. Seems the `m.workerProjectMap` residents always with N entries, where N is the number of active job masters.

Reply:
It's used when we call 'CancelJob'.
I want to wait the 'job management' pr.